src/main/java/frc/robot/subsystems/ball/shooter/Shooter.java

```diff
@@ -1,9 +1,12 @@
 package frc.robot.subsystems.ball.shooter;
 
 import ca.team3161.lib.robot.subsystem.Subsystem;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
 
 public interface Shooter extends Subsystem{
     void runShooter();
     void stopShooter();
+    Value getHatch();
```

Comment:
I would prefer to abstract the `Value` type away and not expose it in this interface. For API consumers, setting the hatch to `Forward` doesn't really have meaning - what they want is more like "open" and "closed", but the correlation between "forward/reverse/off" and "open/closed" is not really obvious.

Reply:
You mean something like an enum that handles that?

Reply:
It could be an enum. It could also be a `boolean` with appropriate names. `boolean isHatchOpen()` and `setHatchOpen(boolean state)` are pretty clear and don't need an enum definition.

Reply:
Ah true, I can just add a couple if statements in there and then return or set the according value.

Reply:
Yea. That logic needs to be handled somewhere regardless, so my point is it makes sense for this implementation to handle that logic and hide it away as an implementation detail, rather than forcing interface consumers to figure out what a DoubleSolenoid.Value means for a hatch mech

Reply:
Also, consider what happens if the hatch mech hardware gets changed out from a DoubleSolenoid to a regular single Solenoid, or if it becomes a motorized LinearActuator, or anything else. Those are all implementation details, and it shouldn't matter to interface consumers that that changed, because the subsystem still exposes the same public functionality. But none of those things would use a DoubleSolenoid.Value.